src/rust/en.omegascans/src/parser.rs

```diff
@@ -0,0 +1,213 @@
+use aidoku::{
+	error::Result, helpers::uri::encode_uri_component, prelude::format, std::net::HttpMethod,
+	std::net::Request, std::String, std::Vec, Chapter, Filter, FilterType, Manga,
+	MangaContentRating, MangaPageResult, MangaStatus, MangaViewer, Page,
+};
+
+const BASE_API_URL: &str = "https://api.omegascans.org";
+
+pub fn parse_manga_list(base_url: String, filters: Vec<Filter>) -> Result<MangaPageResult> {
+	let mut search_query = String::new();
+	let mut status = String::new();
+	let mut genres = String::new();
+
+	for filter in filters {
+		match filter.kind {
+			FilterType::Title => {
+				if let Ok(filter_value) = filter.value.as_string() {
+					search_query.push_str(&encode_uri_component(filter_value.read()));
+				}
+			}
+			FilterType::Genre => {
+				if let Ok(filter_id) = filter.object.get("id").as_string() {
+					match filter.value.as_int().unwrap_or(-1) {
+						1 => {
+							genres.push_str(filter_id.read().as_str());
+							genres.push(',');
+						}
+						_ => continue,
+					}
+				}
+			}
+			FilterType::Select => match filter.name.as_str() {
+				"Status" => {
+					status = match filter.value.as_int().unwrap_or(-1) {
+						0 => String::from("All"),
+						1 => String::from("Ongoing"),
+						2 => String::from("Hiatus"),
+						3 => String::from("Dropped"),
+						4 => String::from("Completed"),
+						5 => String::from("Cancelled"),
+						_ => continue,
+					};
+				}
+				_ => continue,
+			},
+			_ => continue,
+		}
+	}
+
+	if !genres.is_empty() {
+		genres.pop();
+	}
+
+	let url = format!("{}/query?query_string={}&series_status={}&order=desc&orderBy=total_views&series_type=Comic&page=1&perPage=1000&tags_ids=[{}]", BASE_API_URL, search_query, status, genres);
```

Comment:
wait I just realized, why on earth are you loading 1000 manga in a single page? use the `page` parameter and do like 20 per page instead, same goes for listing

Reply:
Oh sorry, that was previously before I knew about the page thing. I did it for webtoonxyz but not for omegascans lol